stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1507,6 +1511,69 @@ func _TFCOpSetAttrStringArray(_ op: CTFEOp,
   }
 }
 
+/// A TensorFlow device kind.
+public enum DeviceKind : UInt {
+  /// Default device.
+  case `default` = 0
+  /// Central processing units.
+  case cpu = 1
+  /// Graphics processing units.
+  case gpu = 2
+  // TODO: TPU?
+}
+
+internal extension _ExecutionContext {
+  var currentDeviceKind: DeviceKind {
+    return sync {
+      deviceScopes.last ?? .default
+    }
+  }
+
+  @usableFromInline
+  func pushDeviceScope(_ kind: DeviceKind) {
+    sync {
+      deviceScopes.append(kind)
+    }
+  }
+
+  @usableFromInline
+  func popDeviceScope() {
+    sync {
+      internalConsistencyCheck(deviceScopes.popLast() != nil)
+    }
+  }
+}
+
+/// Execute a closure, making TensorFlow operations run on a specific kind of
+/// device unless specified otherwise.
+///
+/// - Parameters:
+///   - kind: A kind of device to run TensorFlow operations on.
+///   - body: A closure whose TensorFlow operations are to be executed on the
+///   specified kind of device.
+@inlinable
+public func withDevice<R>(_ kind: DeviceKind, perform body: () throws -> R) rethrows -> R {
```

Comment:
This exceeds 80 columns.

Reply:
I remember being told it was 100 columns some time before. Is this not the case anymore?

Reply:
To clarify, it's 100 columns for Swift code outside of the compiler repo, per https://google.github.io/swift/. This is part of the compiler repo, so it follows the stdlib convention which is 80 columns.